plugins/jenkins/app/models/samson/jenkins.rb

```diff
@@ -30,17 +52,142 @@ def initialize(job_name, deploy)
       @deploy = deploy
     end
 
+    def jenkins_job_config
+      conf = Rails.cache.fetch(job_name + "_conf", expires_in: 1.day, race_condition_ttl: 5.minute) do
+        client.job.get_config(job_name)
+      end
+      Nokogiri::XML(conf)
+    end
+
+    def configure_job_config
```

Comment:
some details about the reasoning would be nice here ... why do we only post if there is a description or missing params ...